_maps/RandomRuins/BeachRuins/beach_treasure_cove.dmm

```diff
@@ -0,0 +1,3215 @@
+/area/ruin/beach/treasure_cove)
+"GC" = (
+/obj/structure/spacevine,
+/turf/open/water/beach,
+/area/ruin/unpowered)
+"Hf" = (
+/obj/effect/turf_decal/siding/wood{
+	dir = 1
+	},
+/obj/effect/decal/cleanable/dirt/dust,
+/turf/open/floor/wood{
+	icon_state = "wood-broken2"
+	},
+/area/ruin/beach/treasure_cove)
+"HP" = (
+/obj/effect/turf_decal/siding/wood{
+	dir = 4
+	},
+/turf/open/water/beach/deep,
+/area/ruin/unpowered)
+"Iv" = (
+/turf/template_noop,
+/area/template_noop)
+"IQ" = (
+/obj/effect/turf_decal/siding/wood/corner{
+	dir = 1
+	},
+/obj/effect/decal/cleanable/dirt/dust,
+/turf/open/floor/wood{
+	icon_state = "wood-broken"
+	},
+/area/ruin/beach/treasure_cove)
+"Jd" = (
+/obj/structure/flora/grass/jungle,
+/obj/structure/railing/wood{
+	layer = 3.1;
+	dir = 8
+	},
+/turf/open/floor/plating/dirt/jungle,
+/area/ruin/unpowered)
+"Je" = (
+/obj/effect/turf_decal/siding/wood,
+/obj/structure/curtain/bounty,
+/obj/structure/fluff/hedge,
+/turf/open/floor/wood,
+/area/ruin/beach/treasure_cove)
+"Jh" = (
+/obj/structure/destructible/tribal_torch{
+	pixel_x = 9;
+	pixel_y = 2
+	},
+/turf/open/floor/plating/dirt/jungle,
+/area/ruin/unpowered)
+"Jm" = (
+/obj/effect/turf_decal/weather/dirt{
+	dir = 4
+	},
+/turf/open/water/beach,
+/area/ruin/unpowered)
+"JM" = (
+/obj/effect/turf_decal/weather/dirt{
+	dir = 10
+	},
+/obj/structure/flora/tree/jungle/small{
+	pixel_x = -36
+	},
+/obj/structure/flora/grass/jungle{
+	pixel_x = 2;
+	pixel_y = -15
+	},
+/obj/effect/decal/cleanable/dirt/dust,
+/turf/open/floor/plating/dirt/jungle/wasteland,
+/area/ruin/unpowered)
+"JQ" = (
+/obj/machinery/autolathe/hacked,
+/obj/item/stack/sheet/metal/twenty{
+	pixel_x = -5;
+	pixel_y = 5
+	},
+/turf/open/floor/plasteel/tech/grid,
+/area/ruin/beach/treasure_cove)
+"Kj" = (
+/obj/effect/turf_decal/siding/wood{
+	dir = 4
+	},
+/obj/effect/decal/cleanable/dirt/dust,
+/turf/open/floor/wood,
+/area/ruin/beach/treasure_cove)
+"Kn" = (
+/obj/structure/chair/wood{
+	dir = 4
+	},
+/obj/effect/decal/cleanable/dirt/dust,
+/turf/open/floor/plating/dirt/jungle/dark/lit,
+/area/ruin/beach/treasure_cove)
+"Ku" = (
+/obj/structure/destructible/tribal_torch/lit{
+	pixel_x = -8;
+	pixel_y = 11
+	},
+/turf/open/floor/plating/dirt/jungle,
+/area/ruin/unpowered)
+"KO" = (
+/obj/effect/decal/cleanable/dirt/dust,
+/obj/effect/decal/cleanable/greenglow,
+/turf/open/floor/plating/dirt/jungle,
+/area/ruin/unpowered)
+"Le" = (
+/obj/structure/destructible/tribal_torch/lit{
+	pixel_x = 12
+	},
+/obj/structure/spacevine,
+/obj/structure/punji_sticks,
+/turf/open/floor/plating/dirt/jungle,
+/area/ruin/unpowered)
+"Lt" = (
+/obj/structure/table/wood/reinforced,
+/obj/item/flashlight/lantern{
+	pixel_x = -6;
+	pixel_y = 10
+	},
+/obj/item/melee/transforming/energy/sword/saber/pirate/red{
+	pixel_y = 8;
+	pixel_x = 10
+	},
+/obj/item/radio{
+	desc = "An old handheld radio. You could use it, if you really wanted to.";
+	icon_state = "radio";
+	name = "old radio";
+	pixel_x = -5;
+	pixel_y = 2
+	},
```

Comment:
you use this a lot, you should make another pr subtyping it and updating your maps to use that subtype.